### PR TITLE
db-refactor-8

### PR DESCRIPTION
**Pull request by freesig**
- Changes to cell to allow space level databases.
- Some functions are moved to the `space.rs` file as they don't make sense at the cell level any more.

- Depends on #1122 
- Followed by #1126

### CODE REVIEW

File: crates/holochain/src/conductor/cell.rs [hunk 1]
```diff
@@ -980,38 +780,41 @@ impl Cell {
...
-        let conductor_api = self.conductor_api.clone();
+        let conductor_handle = self.conductor_handle.clone();
         let signal_tx = self.signal_broadcaster().await;
         let ribosome = self.get_ribosome().await?;
-        let invocation = ZomeCallInvocation::from_interface_call(conductor_api.clone(), call).await;
+        let invocation =
+            ZomeCallInvocation::from_interface_call(self.conductor_api.clone(), call).await;
 
         let args = CallZomeWorkflowArgs {
+            cell_id: self.id.clone(),
```
Comment (thedavidmeister) [inline crates/holochain/src/conductor/cell.rs:813]: note this goes away but maybe you should merge your stuff first

**Review by thedavidmeister — APPROVED** [review]

File: crates/holochain/src/conductor/cell.rs (hunk 1 above)
Comment (freesig) [reply crates/holochain/src/conductor/cell.rs:813]: H,, I wonder if this is a problem? How does the host function find out the agent without this?

File: crates/holochain/src/conductor/cell.rs [hunk 2]
```diff
@@ -1110,12 +914,13 @@ impl Cell {
...
-            .remove()
-            .await
-            .map_err(|e| CellError::Cleanup(e.to_string(), path))?;
+        //TODO "This will need to be reference counted if we really want to delete it"
+        // let path = self.authored_env.path().clone();
+        // // Delete directory
+        // self.authored_env
+        //     .remove()
+        //     .await
+        //     .map_err(|e| CellError::Cleanup(e.to_string(), path))?;
```
Comment (neonphog) [inline crates/holochain/src/conductor/cell.rs:923]: o_O - would we ever want to delete this?

File: crates/holochain/src/conductor/manager/mod.rs [hunk 3]
```diff
@@ -120,6 +129,9 @@ pub enum TaskOutcome {
     /// Either pause or disable all apps which contain the problematic Cell,
     /// depending upon the specific error.
     StopApps(CellId, Box<ManagedTaskError>, String),
+    /// Either pause or disable all apps which contain the problematic Dna,
+    /// depending upon the specific error.
+    StopAppsWithDna(Arc<DnaHash>, Box<ManagedTaskError>, String),
```
Comment (neonphog) [inline crates/holochain/src/conductor/manager/mod.rs:134]: :+1:

**Review by neonphog — APPROVED** [review]
Nice, I like some of the cleanup that comes with moving stuff into the shared "space".

File: crates/holochain/src/conductor/cell.rs (hunk 2 above)
Comment (freesig) [reply crates/holochain/src/conductor/cell.rs:923]: resolved in `5bb543b56142472e18bccba2c54ca17179ad0d14`

**Pull request closed without merging**